### PR TITLE
Add cppcheck and make fully C89 compatible

### PR DESCRIPTION
**Pull request by Zifzaf**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,21 +3,28 @@ version: 2
...
     steps:
       - checkout
       - run: make style-check
 
+  run-cppcheck:
+    docker:
+      - image: sensirion/embedded-ci:2.4.0
+    steps:
+      - checkout
+      - run: cppcheck --std=c++11 --language=c++ --error-exitcode=1 --enable=warning,style,performance,portability -i i2c/sample-implementations/ -i shdlc/sample-implementations/ .
```
Comment (rnestler) [inline .circleci/config.yml:23]: It's not c++, so I'd change the language to `c` and the std to `c89` or `c99`. I'm not sure if we should try to support `c89` for really old embedded compiler or just go for `c99`.

Comment (Zifzaf) [reply .circleci/config.yml]: Choose `c99`.

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,21 +3,28 @@ version: 2
...
     steps:
       - checkout
       - run: make style-check
 
+  run-cppcheck:
+    docker:
+      - image: sensirion/embedded-ci:2.4.0
+    steps:
+      - checkout
+      - run: cppcheck --std=c99 --language=c++ --error-exitcode=1 --enable=warning,style,performance,portability -i i2c/sample-implementations/ -i shdlc/sample-implementations/ .
```
Comment (rnestler) [inline .circleci/config.yml:23]: It's still c++?

File: i2c/sample-implementations/GPIO_bit_banging/sample-implementations/Atmel_SAMD2_series/sensirion_i2c_gpio.c [hunk 3]
```diff
@@ -129,5 +129,5 @@ uint8_t sensirion_i2c_gpio_SCL_read(void) {
  * @param useconds the sleep time in microseconds
  */
 void sensirion_i2c_gpio_sleep_usec(uint32_t useconds) {
-    // IMPLEMENT
+    /* IMPLEMENT */
```
Comment (abrauchli) [inline i2c/sample-implementations/GPIO_bit_banging/sample-implementations/Atmel_SAMD2_series/sensirion_i2c_gpio.c:132]:
```suggestion
    /* TODO: IMPLEMENT */
```
might help to highlight it or in conjunction with git grep

**Comment by rnestler** [comment]
Needs rebasing. Probably you didn't start with an up to date driver-generator/main branch.

**Review by rnestler — APPROVED** [review]
Apart from the missing rebase and the feedback from abrauchli it looks fine to me :slightly_smiling_face:

**Review by rnestler — APPROVED** [review]

**Pull request merged**